packages/datadog-plugin-aws-sdk/src/base.js

```diff
@@ -0,0 +1,152 @@
+'use strict'
+
+const Tags = require('opentracing').Tags
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
+
+class BaseAwsSdkPlugin extends Plugin {
+  static get serviceIdentifier () {
+    const id = this.name.toLowerCase()
+    Object.defineProperty(this, 'serviceIdentifier', {
+      configurable: true,
+      writable: true,
+      enumerable: true,
+      value: id
+    })
+    return id
+  }
+
+  get serviceIdentifier () {
+    const id = this.constructor.serviceIdentifier
+    Object.defineProperty(this, 'serviceIdentifier', {
+      configurable: true,
+      writable: true,
+      enumerable: true,
+      value: id
+    })
+    return id
+  }
+
+  constructor (...args) {
+    super(...args)
+
+    this.addSub(`apm:aws:request:start:${this.serviceIdentifier}`, ({
+      request,
+      operation,
+      awsRegion,
+      awsService,
+      serviceIdentifier
+    }) => {
+      if (!this.isEnabled(request)) {
+        return
+      }
+      const serviceName = this.getServiceName(serviceIdentifier)
+      const childOf = this.tracer.scope().active()
+      const tags = {
+        [Tags.SPAN_KIND]: 'client',
+        'service.name': serviceName,
+        'aws.operation': operation,
+        'aws.region': awsRegion,
+        'aws.service': awsService,
+        'component': 'aws-sdk'
+      }
+      if (this.requestTags) this.requestTags.set(request, tags)
+
+      const span = this.tracer.startSpan('aws.request', { childOf, tags })
+
+      analyticsSampler.sample(span, this.config.measured)
+
+      this.requestInject(span, request)
+
+      const store = storage.getStore()
+
+      this.enter(span, store)
+    })
+
+    this.addSub(`apm:aws:request:complete:${this.serviceIdentifier}`, ({ response }) => {
+      const store = storage.getStore()
+      if (!store) return
+      const { span } = store
+      if (!span) return
+      this.addResponseTags(span, response)
```

Comment:
This is inconsistent with adding request tags.

Reply:
If you'd like me to change something here, please be specific.

Reply:
I'm not sure if this should be changed of the request ones, but this one use a setter and for the request it's a getter like "getRequestTags". It would be more consistent to use the same approach. The setter is probably more flexible.